packages/web/src/pages/edit-collection-page/mobile/EditCollectionPage.tsx

```diff
@@ -55,32 +56,42 @@ const initialFormFields = {
   ...newCollectionMetadata()
 }
 
-type EditPlaylistPageProps = ReturnType<typeof mapStateToProps> &
+type EditCollectionPageParams = {
+  handle: string
+  slug: string
+}
+
+type EditCollectionPageProps = ReturnType<typeof mapStateToProps> &
   ReturnType<typeof mapDispatchToProps>
 
-const g = withNullGuard((props: EditPlaylistPageProps) => {
+const g = withNullGuard((props: EditCollectionPageProps) => {
   const { account } = props
   if (account) return { ...props, account }
 })
 
-const EditPlaylistPage = g(
+const EditCollectionPage = g(
   ({ removeTrack, editPlaylist, orderPlaylist, refreshLineup }) => {
-    const { data, onClose } = useEditPlaylistModal()
-    const { collectionId } = data
-
-    const metadata = useSelector((state) =>
-      getCollection(state, { id: collectionId })
-    )
-    const tracks = useSelector((state) =>
-      getCollectionTracksWithUsers(state, { id: collectionId ?? undefined })
+    const { handle, slug } = useParams<EditCollectionPageParams>()
+    const isAlbum = Boolean(useRouteMatch('/:handle/album/:slug/edit'))
+    const permalink = `/${handle}/${isAlbum ? 'album' : 'playlist'}/${slug}`
+    const dispatch = useDispatch()
+    const history = useHistory()
+
+    const { data: currentUserId } = useGetCurrentUserId({})
+    const { data: collection } = useGetPlaylistByPermalink(
+      {
+        permalink,
+        currentUserId
+      },
+      { disabled: !currentUserId }
     )
-    const messages = getMessages(metadata?.is_album ? 'album' : 'playlist')
 
-    // Close the page if the route was changed
-    useHasChangedRoute(onClose)
+    const { playlist_id, tracks } = collection ?? {}
+
+    const messages = getMessages(collection?.is_album ? 'album' : 'playlist')
 
     const initialMetadata = {
-      ...(metadata as Collection),
+      ...(collection as unknown as Collection),
```

Comment:
Tangent, but I think the need to do this arises from the lack of a cache type mapping in audius-query

The endpoints return api models, but once we cache and return them, we're actually returning the cache models. Been thinking about ways to handle this if you want to discuss sometime